homeassistant/components/adax/__init__.py

```diff
@@ -2,16 +2,38 @@
 
 from __future__ import annotations
 
+import datetime
+import logging
+
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import Platform
 from homeassistant.core import HomeAssistant
 
+from .const import CONNECTION_TYPE, LOCAL
+from .coordinator import AdaxCloudCoordinator, AdaxLocalCoordinator
+
 PLATFORMS = [Platform.CLIMATE]
+SCAN_INTERVAL = datetime.timedelta(seconds=60)
+
+_LOGGER = logging.getLogger(__name__)
 
+type AdaxConfigEntry = ConfigEntry[AdaxCloudCoordinator | AdaxLocalCoordinator]
 
-async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+
+async def async_setup_entry(hass: HomeAssistant, entry: AdaxConfigEntry) -> bool:
     """Set up Adax from a config entry."""
+    if entry.data.get(CONNECTION_TYPE) == LOCAL:
+        coordinator = AdaxLocalCoordinator(hass, entry, update_interval=SCAN_INTERVAL)
+    else:
+        coordinator = AdaxCloudCoordinator(hass, entry, update_interval=SCAN_INTERVAL)
+
+    entry.runtime_data = coordinator
+    _LOGGER.info("Init Adax, coordinator: %s", coordinator)
+    await coordinator.async_config_entry_first_refresh()
+    _LOGGER.info("First refresh complete: %s", coordinator.get_rooms())
```

Comment:
Same, if we need to get data from the coordinator, we should expose that via the diagnostic panel.

```suggestion
```